extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpHandler.java

```diff
@@ -169,9 +170,9 @@ private APIGatewayV2HTTPResponse nettyDispatch(InetSocketAddress clientAddress,
         quarkusHeaders.setContextObject(Context.class, context);
         quarkusHeaders.setContextObject(APIGatewayV2HTTPEvent.class, request);
         quarkusHeaders.setContextObject(APIGatewayV2HTTPEvent.RequestContext.class, request.getRequestContext());
-        final Principal principal = getPrincipal(request);
-        if (principal != null) {
-            quarkusHeaders.setContextObject(Principal.class, principal);
+        final SecurityIdentity identity = LambdaHttpRecorder.identityProvider.create(request);
+        if (identity != null) {
+            quarkusHeaders.setContextObject(SecurityIdentity.class, identity);
```

Comment:
This should just put the `request` object into the headers, and then the security code can be moved into our normal security layer.